src/Fantomas.Core.Tests/AppTests.fs

```diff
@@ -1137,3 +1137,42 @@ let ``function invocation wrapped in parentheses, 2382`` () =
     42
 )
 """
+
+[<Test>]
+let ``comment above single parameter in brackets, 2594`` () =
+    formatSourceString
+        false
+        """
+let doSomething _ _ = ()
+
+let test1 param =
+    doSomething
+        // my comment
+        (param)
+
+let test2 param =
+    doSomething
+        // my comment
+        (param)
+        (param)
+"""
+        config
+    |> prepend newline
+    |> should
+        equal
+        """
+let doSomething _ _ = ()
+
+let test1 param =
+    doSomething
+        // my comment
```

Comment:
@dawedawe I'm still somewhat on the fence about whether we should not try to exclude the trivia before (of the SynExpr.Paren) in the check if the arguments are multiline or not.

Something along the lines of:
- Check if `parenArg` has trivia before, if so indent and next line
- Run expressionFitsOnRestOfLine for expr

This would lead to
```fsharp
    doSomething
        // my comment
        (param)
```

Reply:
Not sure, if I understood your implementation idea correctly.
But I like the change of formatting quite a lot.